website/content/en/docs/concepts/scheduling.md

```diff
@@ -260,6 +260,14 @@ If they all fail, Karpenter will fail to provision the pod.
 Karpenter will backoff and retry over time.
 So if capacity becomes available, it will schedule the pod without user intervention.
 
+#### Preferred Affinities
+
+Karpenter treats preferred affinities as hard affinities in almost all circumstances.
+
+Karpenter interprets preferred affinities as a soft affinity when building up topology requirements for scheduling to a node. Preferred affinities are ignored in regards to topology zones and hard affinities should be used as documented above.
+
+In all other circumstances, Karpenter treats the required and preferred affinities as if they were required affinities. When determining if a pod can schedule on a node (absent topology requirements), or if a pod can be shifted to a new node, soft affinities are treated as if hard affinities. This includes when determining if a node can be safely disrupted, a node with pods that contain unsatisfied preferred affinities will not be disrupted if there are no matching nodes that meet it's preferred requirements.
```

Comment:
What do you mean "contain unsatisfied preferred affinities"? You mean if there is a pod on a node that doesn't satisfy its current preference, we also won't shift the pod to another node that also doesn't satisfy this preference? -- If that's the case that's actually interesting behavior. Not surprising based on how I understand the scheduling behavior, but slightly unexpected. We could probably do better here, so it might be worth opening an issue if this is the case.

Reply:
I think https://github.com/kubernetes-sigs/karpenter/issues/1204 may be an example of this behavior. I haven't been able to locate any special handling for the controlplane label in Karpenter's code.

Reply:
@jonathan-innis I updated the language of this to be clearer and match the behavior around relaxing requirements.

@wmgroot I don't think that's evidence of this, I believe what is happening there is no valid nodeclaim can be generated because of the restricted domain.